admin/app/helpers/spree/admin/session_assets_helper.rb

```diff
@@ -0,0 +1,24 @@
+module Spree
+  module Admin
+    module SessionAssetsHelper
+      def ensure_session_uploaded_assets_uuid
+        session['spree.admin.uploaded_assets.uuid'] ||= SecureRandom.uuid
+      end
+
+      def store_uploaded_asset_in_session(asset)
```

Comment:
@damianlegawiec We send a request for every asset and we have a race condition when trying to store asset IDs in session to display them on a new product. 

I changed it so we create a unique ID for uploaded assets and assign this ID to uploaded assets in metadata. The ID is created when entering the new product page.